playbooks/activate_client_interfaces.yml

```diff
@@ -32,48 +33,78 @@
            item.replace('-', '_')]['active'] }}"
       with_items: "{{ ansible_interfaces }}"
 
-    # A one time copy is made of the network interfaces files after initial
-    # OS install. This is now attempted at the end of install in wait_for_
-    # client_ping.yml, but is re-attempted here for previous installs done
-    # with prior versions of Genesis.
+    - pause:
+        seconds: 5
 
-    - block:
-        - name: Check if deploy t0 interfaces file exists
-          stat: path=/etc/network/interfaces.gen-deploy-t0
-          register: st
-
-        - name: One time copy t0 interfaces file
-          command: "cp /etc/network/interfaces /etc/network/interfaces.gen-deploy-t0"
-          when: not st.stat.exists
+    - name: Insure all interfaces are up
+      command: "ip link set dev {{ item }} up"
+      when:
+        - "{{ item != 'lo' }}"
+        - "{{ ansible_host != hostvars[inventory_hostname]['ansible_' + \
+           item.replace('-', '_')].get('ipv4', {}).get('address') }}"
+      with_items: "{{ ansible_interfaces }}"
+      ignore_errors: yes
 
-        - name: Backup interfaces file
-          command: "cp /etc/network/interfaces /etc/network/interfaces.bak"
+    - pause:
+        seconds: 5
 
-        - name: Write interfaces file for ipv6 auto on all interfaces if Ubuntu
-          template:
-            src: "{{ playbook_dir }}/templates/ipv6Interface_ubuntu.j2"
-            dest: /etc/network/interfaces
-            owner: "root"
-            group: "root"
-            mode: "0644"
-      when: ansible_distribution == 'Ubuntu'
 
-    - name: Write ifcfg files for ipv6 on all interfaces if RHEL/CentOS
-      template:
-        src: "{{ playbook_dir }}/templates/ipv6Interface_rhel.j2"
-        dest: "/etc/sysconfig/network-scripts/ifcfg-{{ item }}"
-        owner: "root"
-        group: "root"
-        mode: "0644"
-      when: (ansible_distribution == 'CentOS') or
-            (ansible_distribution == 'RedHat')
+    # Update switch mac address tables by adding temporary link local
+    # ipv4 address.  This also forces an ipv6 link local setup which
+    # drives traffic to the switch.
+    - name: Update switch mac address tables
+      command: "sudo ip addr add 169.254.0.2/24 dev {{ item }}"
```

Comment:
I'm not sure if there's any _real_ benefit, but the preferred way elevate privilege in Ansible is to use "become". e.g.: `become: true`

Reply:
Also, we're actually running as root on the client nodes anyways, so you shouldn't even need to escalate privilege.

Reply:
changed to become: true